payments_service/braintree/views.py

```diff
@@ -59,6 +60,25 @@ def post(self, request):
         if not form.is_valid():
             return error_400(response=form.errors)
 
+        if settings.ALLOW_REPEAT_PAYMENTS:
+            log.warn('not protecting against repeat subscriptions')
+        else:
+            product = self.api.generic.product.get_object_or_404(
+                public_id=form.cleaned_data['plan_id'],
+            )
+            # Check if the user is already subscribed to this plan.
+            result = self.api.braintree.mozilla.subscription.get(
+                paymethod__braintree_buyer__buyer=self.request.user.pk,
+                seller_product=product['resource_pk'],
```

Comment:
yeah, that would be cool. I filed: https://github.com/mozilla/solitude/issues/466